deployability/modules/allocation/vagrant/instance.py

```diff
@@ -109,61 +126,71 @@ def ssh_connection_info(self) -> ConnectionInfo:
+                    if not Path(tmp_port_file).exists():
+                        port = VagrantUtils.get_port(self.remote_host_parameters)
+                        cmd = f"sudo /usr/bin/ssh -i /Users/jenkins/.ssh/localhost -L {server_ip}:{port}:{ip}:22 -N 127.0.0.1 -f"
+                        VagrantUtils.remote_command(cmd, self.remote_host_parameters)
+                        with open(tmp_port_file, 'w') as f:
+                            f.write(port)
+                    else:
+                        with open(tmp_port_file, 'r') as f:
+                            port = f.read()
+                    ssh_config['port'] = port
                 else:
                     with open(tmp_port_file, 'r') as f:
                         port = f.read()
-                ssh_config['port'] = port
+                    ssh_config['port'] = port
+                ssh_config['hostname'] = server_ip
+                ssh_config['user'] = 'vagrant'
+                ssh_config['password'] = 'vagrant'
+            elif self.platform == 'windows':
+                for key, pattern in patterns.items():
+                    match = re.search(pattern, output)
+                    if match and key == 'hostname':
+                        ip = match.group(1).strip()
+                ssh_config['hostname'] = ip
+                ssh_config['port'] = 3389
+                ssh_config['user'] = 'vagrant'
+                ssh_config['password'] = 'vagrant'
             else:
-                with open(tmp_port_file, 'r') as f:
-                    port = f.read()
-                ssh_config['port'] = port
-            ssh_config['hostname'] = server_ip
-            ssh_config['user'] = 'vagrant'
-            ssh_config['password'] = 'vagrant'
-        elif self.platform == 'windows':
-            for key, pattern in patterns.items():
-                match = re.search(pattern, output)
-                if match and key == 'hostname':
-                    ip = match.group(1).strip()
-            ssh_config['hostname'] = ip
-            ssh_config['port'] = 3389
-            ssh_config['user'] = 'vagrant'
-            ssh_config['password'] = 'vagrant'
-        else:
-            for key, pattern in patterns.items():
-                match = re.search(pattern, output)
-                if match:
-                    ssh_config[key] = str(match.group(1)).strip("\r")
-                else:
-                    logger.error(f"Couldn't find {key} in ssh-config")
-                    return None
-            if self.credentials:
-                ssh_config['private_key'] = str(self.credentials.key_path)
+                for key, pattern in patterns.items():
+                    match = re.search(pattern, output)
+                    if match:
+                        ssh_config[key] = str(match.group(1)).strip("\r")
+                    else:
+                        logger.error(f"Couldn't find {key} in ssh-config")
+                        return None
```

Comment:
Sometimes it can be cleaner if you just validate with a conditional the negative option. In this case, when there is no match, you can just raise the error log and return, note that the `return` in python by default retrieves a None, anyway I think its OK to specify it, so anyone reads it and knows its returning a None

```suggestion
                    if not match:
                        logger.error(f"Couldn't find {key} in ssh-config")
                        return None
                    ssh_config[key] = str(match.group(1)).strip("\r")

```

Reply:
Done dc8184857